### PR TITLE
codigo

### PR DESCRIPTION
**Pull request by MonteiroVideo**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,1146 @@
...
+					},
+				],
+				callback: async (event) => {
+					const path = await this.parseVariablesInString(event.options.path)
+					const int = await this.parseVariablesInString(event.options.int)
+
+					sendOscMessage(path, [
+						{
+							type: 'i',
+							value: parseInt(int),
```
Comment (dnmeid) [inline main.js:701]: What happens when the user enters `-149588357.2`? Or `this is not a number`? Regex on option does not validate input, it just gives a visual hint.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,1146 @@
...
+						},
+					])
+				},
+			},
+			send_ant: {
+				name: 'Anterior',
```
Comment (dnmeid) [inline main.js:707]: All the same as above

File: main.js [hunk 3]
```diff
@@ -0,0 +1,1146 @@
...
+
+					sendOscMessage(path, [
+						{
+							type: 'i',
+							value: parseInt(int),
+						},
+					])
+				},
+			},
+			send_prox: {
```
Comment (dnmeid) [inline main.js:737]: Again all the same as above

File: main.js [hunk 4]
```diff
@@ -0,0 +1,1146 @@
...
+
+					sendOscMessage(path, [
+						{
+							type: 'i',
+							value: parseInt(int),
+						},
+					])
+				},
+			},
+			send_exit: {
```
Comment (dnmeid) [inline main.js:768]: And again

File: main.js [hunk 5]
```diff
@@ -0,0 +1,1146 @@
...
+
+					sendOscMessage(path, [
+						{
+							type: 'i',
+							value: parseInt(int),
+						},
+					])
+				},
+			},
+			send_exitV: {
```
Comment (dnmeid) [inline main.js:799]: And again again

File: main.js [hunk 6]
```diff
@@ -0,0 +1,1146 @@
...
+
+					sendOscMessage(path, [
+						{
+							type: 'i',
+							value: parseInt(int),
+						},
+					])
+				},
+			},
+			send_video1: {
```
Comment (dnmeid) [inline main.js:830]: History repeating, I hope you got the point.

File: presets.js
Comment (dnmeid) [inline presets.js:1]: Please also use english for presets

**Review by dnmeid — CHANGES_REQUESTED** [review]
At this stage this module is just a prefilled generic OSC sender. It provides only a tiny advantage over the generic OSC module.
I still don't see how your software does provide any advantage over the existing PPT control plugins. They do have a lot more features and useful feedback. But that is your decision. From my side the code needs to be error free and the module should be usable for the average user and at the moment it is simply bad design.

**Comment by MonteiroVideo** [comment]
ITS OK

File: companion/manifest.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+	"id": "monteiro-pptvideoosc",
+	"name": "monteiro-pptvideoosc",
+	"shortname": "monteiro-pptvideoosc",
+	"description": "Perform operations on separate machines.",
```
Comment (dnmeid) [inline companion/manifest.json:5]: Please use a more precise and descriptive description. "operations" can be anything, "machines" can be anything

File: companion/manifest.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
+{
+	"id": "monteiro-pptvideoosc",
+	"name": "monteiro-pptvideoosc",
+	"shortname": "monteiro-pptvideoosc",
```
Comment (dnmeid) [inline companion/manifest.json:4]: the shortname is the same as the name. it should be shorter, just the minimum to recognize the connection

File: companion/manifest.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+	"runtime": {
+		"type": "node18",
+		"api": "nodejs-ipc",
+		"apiVersion": "0.0.0",
+		"entrypoint": "../main.js"
+	},
+	"legacyIds": [],
+	"manufacturer": "Monteiro",
+	"products": [ "PPT-VIDEO-OSC" ],
+	"keywords": [POWERPOINT, VIDEO, TIMER]
```
Comment (dnmeid) [inline companion/manifest.json:25]: There is no timer related handling in the module, this is a misleading keyword.

File: companion/HELP.md [hunk 10]
```diff
@@ -0,0 +1,4 @@
+## PPT VIDEO
+All the presets are done, but if you have any questions, send me a message on Instagram. Monteiro.audiovisual
```
Comment (dnmeid) [inline companion/HELP.md:2]: The help file should give help about the module, the configuration, the actions, the feedbacks and so on. If I read this, I get no help. And I have questions: what is this connection for? I can't see its purpose.

File: main.js [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+	updateActions() {
+		const sendOscMessage = (path, args) => {
+			this.log('debug', `Sending OSC ${this.config.host}:${this.config.port} ${path}`)
+			this.log('debug', `Sending Args ${JSON.stringify(args)}`)
+			this.oscSend(this.config.host, this.config.port, path, args)
+		}
+		const actionDefinitions = {};
+
+		// Adicionando send_ppt
+		for (let i = 1; i <= 20; i++) {
```
Comment (dnmeid) [inline main.js:53]: You still completely don't understand, what you are doing wrong and I really don't know how I can explain it in even more detail.
1. Do not make 20 different actions if there is only one parameter changing. Instead make this _**one**_ action where the changing parameter is an _**option**_.
2. There is completely no use in your 'COMANDO' in beinig a user editable textinput. Actually this makes all your actions the same action.

File: main.js [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+						label: 'COMANDO',
+						id: 'path',
+						default: defaultPath,
+						useVariables: true,
+					},
+					{
+						type: 'textinput',
+						label: 'Value',
+						id: 'int',
+						default: 1,
```
Comment (dnmeid) [inline main.js:71]: Default value for a textinput needs to be a string

File: main.js [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+							type: 'i',
+							value: parseInt(int),
+						},
+					]);
+				},
+			};
+		}
+
+		// Adicionando send_int
+		actionDefinitions[`send_int`] = {
```
Comment (dnmeid) [inline main.js:91]: This is still copy/paste from the OSC module

File: main.js [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+					{
+						type: 'i',
+						value: parseInt(int),
+					},
+				]);
+			},
+		};
+
+		// Adicionando send_video
+		for (let i = 1; i <= 10; i++) {
```
Comment (dnmeid) [inline main.js:124]: Same as above

File: main.js [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+		];
+
+		commands.forEach((command, index) => {
+			const { name, defaultPath } = command;
+			actionDefinitions[`send_${name.toLowerCase().replace(/\s/g, "_")}`] = {
+				name: name,
+				options: [
+					{
+						type: 'textinput',
+						label: 'COMANDO',
```
Comment (dnmeid) [inline main.js:176]: Good use of forEach here, but the 'COMANDO' option should not be present at all, it is not needed.

File: package.json [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+	"version": "1.0.0",
+	"main": "main.js",
+	"scripts": {
+		"format": "prettier -w .",
+		"package": "companion-module-build"
+	},
+	"license": "MIT",
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/MonteiroVideo/companion-module-monteiro-pptvideoosc.git"
```
Comment (dnmeid) [inline package.json:12]: Update url

File: presets.js [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+                text: `VIDEO\n${index}`,
+                size: 'auto',
+                color: combineRgb(255, 255, 255),
+                bgcolor: combineRgb(25, 25, 200),
+            },
+            steps: [
+                {
+                    down: [
+                        {
+                            actionId: 'send_int',
```
Comment (dnmeid) [inline presets.js:52]: You are using the generic send_int action, where you have a dedicated action

File: presets.js [hunk 18]
```diff
@@ -0,0 +1,217 @@
...
+            text: 'ESC',
+            size: 'auto',
+            color: combineRgb(255, 255, 255),
+            bgcolor: combineRgb(0, 0, 120),
+        },
+        steps: [
+            {
+                down: [
+                    {
+                        actionId: 'send_int',
```
Comment (dnmeid) [inline presets.js:95]: You are using the generic send_int action for a process, which doesn't need an integer at all and which should have its own action.

File: presets.js [hunk 19]
```diff
@@ -0,0 +1,217 @@
...
+
+module.exports = function (self) {
+    const presets = {};
+
+    // Função para criar presets de PPT
+    const createPptPreset = (index) => {
+        return {
+            type: 'button',
+            category: 'PPT',
+            name: 'pr_init',
```
Comment (dnmeid) [inline presets.js:11]: All your presets have the same meaningless name.

File: presets.js [hunk 20]
```diff
@@ -0,0 +1,217 @@
...
+            text: '<\nAnterior',
+            size: 'auto',
+            color: combineRgb(255, 255, 255),
+            bgcolor: combineRgb(255, 0, 0),
+        },
+        steps: [
+            {
+                down: [
+                    {
+                        actionId: 'send_int',
```
Comment (dnmeid) [inline presets.js:122]: same send_int

File: presets.js [hunk 21]
```diff
@@ -0,0 +1,217 @@
...
+            text: '>\nProximo',
+            size: 'auto',
+            color: combineRgb(0, 0, 0),
+            bgcolor: combineRgb(0, 255, 0),
+        },
+        steps: [
+            {
+                down: [
+                    {
+                        actionId: 'send_int',
```
Comment (dnmeid) [inline presets.js:149]: again

File: presets.js [hunk 22]
```diff
@@ -0,0 +1,217 @@
...
+            text: 'Fechar\nPPT',
+            size: 'auto',
+            color: combineRgb(255, 255, 255),
+            bgcolor: combineRgb(0, 0, 0),
+        },
+        steps: [
+            {
+                down: [
+                    {
+                        actionId: 'send_int',
```
Comment (dnmeid) [inline presets.js:176]: again

File: presets.js [hunk 23]
```diff
@@ -0,0 +1,217 @@
...
+            text: 'Fechar\nVideo',
+            size: 'auto',
+            color: combineRgb(255, 255, 255),
+            bgcolor: combineRgb(255, 25, 25),
+        },
+        steps: [
+            {
+                down: [
+                    {
+                        actionId: 'send_int',
```
Comment (dnmeid) [inline presets.js:203]: again

**Review by dnmeid — CHANGES_REQUESTED** [review]
It's not ok

**Pull request closed without merging**